scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,300 @@
+    x[infs] = np.sign(x[infs])*CONST_INF
+    return x
+
+
+def _linprog_highs(lp, solver, time_limit=None, presolve=True,
+                   disp=False, maxiter=None, autoscale=False,
+                   dual_feasibility_tolerance=None,
+                   dual_objective_value_upper_bound=None,
+                   message_level=1,
+                   primal_feasibility_tolerance=None,
+                   simplex_crash_strategy=None,
+                   simplex_dual_edge_weight_strategy=None,
+                   simplex_primal_edge_weight_strategy=None,
+                   simplex_strategy=None,
+                   simplex_update_limit=None,
+                   **unknown_options):
+    r"""
+    Solve the following linear programming problem using one of the HiGHS
+    solvers:
+
+    .. math::
+
+        \min_x \ & c^T x \\
+        \mbox{such that} \ & A_{ub} x \leq b_{ub},\\
+        & A_{eq} x = b_{eq},\\
+        & l \leq x \leq u ,
+
+    where :math:`x` is a vector of decision variables; :math:`c`,
+    :math:`b_{ub}`, :math:`b_{eq}`, :math:`l`, and :math:`u` are vectors; and
+    :math:`A_{ub}` and :math:`A_{eq}` are matrices.
+
+    Informally, that's:
+
+    minimize::
+
+        c @ x
+
+    such that::
+
+        A_ub @ x <= b_ub
+        A_eq @ x == b_eq
+        lb <= x <= ub
+
+    Note that by default ``lb = 0`` and ``ub = None`` unless specified with
+    ``bounds``.
+
+    Parameters
+    ----------
+    lp :  _LPProblem
+        A ``scipy.optimize._linprog_util._LPProblem`` ``namedtuple``.
+    solver : "ipm" or "simplex"
+        Which HiGHS solver to use.
+
+    Options
+    -------
+    maxiter : int
+        The maximum number of iterations to perform in either phase.
+    disp : bool
+        Set to ``True`` if indicators of optimization status are to be printed
+        to the console each iteration; default ``False``.
+    time_limit : float
+        The maximum time in seconds allotted to solve the problem; default is
+        unlimited.
+    presolve : bool
+        Set to ``False`` if presolve is to be disabled; default ``True``.
+    autoscale : bool
+        Set to ``True`` to automatically perform equilibration.
+        Consider using this option if the numerical values in the
+        constraints are separated by several orders of magnitude.
+        Default: ``False``.
+    unknown_options : dict
+        Optional arguments not used by this particular solver. If
+        `unknown_options` is non-empty, a warning is issued listing all
+        unused options.
+    dual_feasibility_tolerance : double
+        Dual feasibility tolerance
+    dual_objective_value_upper_bound : double
+        Upper bound on objective value for dual simplex:
+        algorithm terminates if reached
+    message_level : int {0, 1, 2, 4}
+        Verbosity level, corresponds to:
+
+            ``0``: ML_NONE
+
+            ``1``: ML_VERBOSE
+
+            ``2``: ML_DETAILED
+
+            ``4``: ML_MINIMAL
```

Comment:
Option  levels 0,1,2,4 - could that mean there is a missing option?

Reply:
3 is not a valid option, however after reviewing the [enum definition](https://github.com/ERGO-Code/HiGHS/blob/9318ac7ad706a09b35691e21288b08889f322b77/src/lp_data/HConst.h#L47), it appears that `ML_ALWAYS` with a value of `7` is missing

Reply:
:+1: just making sure. Thank you for all the work you've put in here. It's looking really promising!